src/IO_Strings.f90

```diff
@@ -1629,4 +1630,150 @@ FUNCTION str_SDK_nDecimal(r,nDecimal) RESULT(string)
 
     ENDFUNCTION str_SDK_nDecimal
 !
+!------------------------------------------------------------------------------
+!> @brief This subroutine writes a 2-D array of strings to an output file as a
+!>        formatted table.
+!> @param funit The output unit of the file where the table is written.
+!> @param tablevals The 2-D array of strings to write.
+!>
+!> Note: The default delimiter for writing multi-row strings to a single cell is
+!>       the '; ', where the table functions check for the ';', and the getField
+!>       routine looks for the whitespace.  Surround strings with quotes to have
+!>       them as a single field.
+!>
+    SUBROUTINE stringTableToLines(tablevals,lines)
+      CHARACTER(LEN=*),PARAMETER :: myName='stringTableToLines'
+      TYPE(StringType),INTENT(IN) :: tablevals(:,:)
+      TYPE(StringType),ALLOCATABLE,INTENT(OUT) :: lines(:)
+      LOGICAL(SBK) :: hasString
+      INTEGER(SIK),ALLOCATABLE :: maxcolsize(:),maxrowsize(:)
+      INTEGER(SIK) :: i,j,k,rightpad,ierr,nrow
+      TYPE(StringType) :: plstr,rowstr,field
+
+      !Loop over the string array and write the data and make it pretty.
+      ALLOCATE(maxcolsize(SIZE(tablevals,DIM=1)))
+      ALLOCATE(maxrowsize(SIZE(tablevals,DIM=2)))
+      maxcolsize=0
+      maxrowsize=0
+      CALL getTableBounds(tablevals,maxcolsize,maxrowsize)
+
+      !Three for the top, first row, and last row separator strings.
+      ALLOCATE(lines(SUM(maxrowsize)+3))
+      rowstr=getRowStr(maxcolsize)
+      nrow=1
+      lines(nrow)=rowstr
+      rightpad=0
+
+      !
+      !Loop over the rows
+      DO j=1,SIZE(tablevals,DIM=2)
```

Comment:
So the indexing is (column,row)?  Seems backwards...

Reply:
I was thinking x position and y position in the table.

Reply:
Ok.  I guess that's fine, just opposite of how I was thinking of it.  It looks like the memory is still accessed in optimal order, so it's ok as is.

I do think it would be good to note the indexing in the header comments for this routine though: "The first index of tablevals is the x position (from left to right) in the table and the second index is the y position (from top to bottom) in the table".  This should help alleviate any confusion by other users.

Reply:
Agreed, that's a good note to have.